File: platform/android/overlay/TMessagesProj/src/main/java/app/nebulagram/ui/NebulaRetentionPolicy.java
package app.nebulagram.ui;

/** Explicit opt-ins apply only to received content, never locally initiated deletion. */
public final class NebulaRetentionPolicy {
    private NebulaRetentionPolicy() { }
    public static boolean allowed(boolean enabled, boolean secret, boolean expiring,
            boolean saveSecret, boolean saveExpiring, boolean protectedContent,
            boolean incoming, boolean service, boolean validId) {
        return enabled && incoming && !service && validId && !protectedContent
                && (!secret || saveSecret) && (!expiring || saveExpiring);
    }
}

File: platform/ios/NebulaSettingsContract/Sources/NebulaSettingsContract/NebulaRetentionPolicy.swift
import Foundation

public enum NebulaRetentionPolicy {
    public static func allowed(enabled: Bool, secret: Bool, expiring: Bool, saveSecret: Bool, saveExpiring: Bool, protectedContent: Bool, incoming: Bool, service: Bool, validId: Bool) -> Bool {
        enabled && incoming && !service && validId && !protectedContent && (!secret || saveSecret) && (!expiring || saveExpiring)
    }
}
